helper/call.go

```diff
@@ -361,6 +363,28 @@ func (c *ContractCaller) DecodeValidatorStakeUpdateEvent(receipt *ethTypes.Recei
 	return event, nil
 }
 
+// DecodeValidatorJoinEvent represents validator staked event
+func (c *ContractCaller) DecodeValidatorJoinEvent(receipt *ethTypes.Receipt, logIndex uint64) (*stakinginfo.StakinginfoStaked, error) {
+	event := new(stakinginfo.StakinginfoStaked)
+
+	found := false
+	for _, vLog := range receipt.Logs {
+		if uint64(vLog.Index) == logIndex {
```

Comment:
The log index would be fixed for ValidatorJoin log, right?

Reply:
If yes, we can remove one parameter.

Reply:
No. `LogIndex` is against the block not the tx. The same misconception introduced this bug in `ValidatorJoin` command.

Reply:
yep: https://github.com/ethereum/go-ethereum/blob/master/core/types/log.go#L50